kernel/vmemitcodes.ml

```diff
@@ -513,8 +580,7 @@ let to_memory fv code =
     reloc_info = RelocTable.create 91;
   } in
   emit env code [];
-  (** Later uses of this string are all purely functional *)
-  let code = Bytes.sub_string env.out_buffer 0 env.out_position in
+  let code = compress_code env.out_buffer env.out_position in
```

Comment:
I think that this is the wrong place to perform the compression, because when evaluating non-constants in the VM it will be immediately decompressed right after. This was one of the reasons why I introduced a VM library data structure on disk, the compression should occur around when building it.

Reply:
Good point. That said, I would not be surprised if non-constants were to be rather short, bytecode-wise. (They might be huge, term-wise, especially when doing computational reflection, but most of that size will come from block values whose associated bytecode is trivial.) So, the cost of compressing/decompressing might be in the noise and not worth complicating the code. For example, heavy `vm_compute` users like Four-Color did not suffer from any slowdown.